.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release & Publish to NPM
+
+on:
+  pull_request:
+  push:
+    branches: [main]
```

Comment:
I am not sure I guess you don't want to always execute a full release when a merge on master is done, probably something like an insider release with a hash could be an idea. 

We have implemented a workflow in our project that creates an insider-release after every merge to main. It's far from perfect but could be an option: 
https://github.com/six-group/six-webcomponents/blob/main/.github/workflows/insider-release.yml